### PR TITLE
Make logger configurable and log socket data.

### PR DESCRIPTION
**Pull request by dblock**
This is a replacement for https://github.com/dblock/slack-ruby-client/pull/50.

### CODE REVIEW

**Comment by dblock** [comment]
@mikz FYI I finished what you started in #50 here.

File: lib/slack/real_time/client.rb [hunk 1]
```diff
@@ -125,6 +127,7 @@ def socket_class
 
       def send_json(data)
         fail ClientNotStartedError unless started?
+        logger.debug("#{self.class.name}#send_data") { data }
```
Comment (mikz) [inline lib/slack/real_time/client.rb]: shouldn't this be `#send_json` ? maybe using `"#{self.class}#${__method__}"` would make it automatic.

File: lib/slack/real_time/concurrency/eventmachine.rb [hunk 2]
```diff
@@ -5,6 +5,26 @@ module Slack
...
+          attr_reader :logger
+          protected :logger
+
+          def initialize(url, protocols = nil, options = {})
+            super
+            @logger = options.delete(:logger) || Slack::RealTime::Config.logger || Slack::Config.logger
+          end
+
+          def parse(data)
+            logger.debug("#{self.class.name}#read") { data }
```
Comment (mikz) [inline lib/slack/real_time/concurrency/eventmachine.rb]: should this one be called parse?

**Comment by mikz** [comment]
This looks great! Good job :+1:

File: lib/slack/real_time/client.rb (hunk 1 above)
Comment (dblock) [inline lib/slack/real_time/client.rb]: :+1: 

Generally I wonder about logging data here, it's a binary blob most of the time. Thoughts?

**Pull request merged**